charts/mageai/templates/deployment.yaml

```diff
@@ -25,6 +26,18 @@ spec:
       serviceAccountName: {{ include "mageai.serviceAccountName" . }}
       securityContext:
         {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      initContainers:
+        - name: wait-for-redis
+          image: alpine
+          env:
+          {{- if .Values.redis.enabled }}
+            - name: REDIS_URL
+              value: redis://{{.Release.Name}}-redis-headless:6379/0
+          {{- else if .Values.redis.url }}
+            - name: REDIS_URL
+              value: {{ .Values.redis.url }}
+          {{- end }}
+          command: ["sh", "-c", "until nc -z -v $(echo $REDIS_URL | cut -d'/' -f3 | cut -d':' -f1) $(echo $REDIS_URL | cut -d'/' -f3 | cut -d':' -f2); do sleep 1; done"]
```

Comment:
Does this command throw error when REDIS_URL is not specified?

Reply:
No, this will wait till the redis server comes up healthy

Reply:
I mean if the redis is not enabled, the REDIS_URL won't be set.

Reply:
thats right, let me take a look what I missed

Reply:
Basically, i have added a check for the init container block to check for Redis, if standalone Redis or custom URL is available it will work else init container wont be executed

Reply:
redis.enabled will spin up new redis server so we need to keep it disabled to use custom URL. if both are not specified REDIS_URL wont be set and init container won't run. if u think the value configuration is confusing I can alter the value file structure